`TestTerminalRequireSessionMfa` flakiness

## Failure

#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds/5b71253c-6aa1-4e81-8dc8-4a3d8528d703?project=ci-account

#### Relevant snippet

```
=== CONT  TestTerminalRequireSessionMfa
    testing.go:1090: TempDir RemoveAll cleanup: unlinkat /tmp/TestTerminalRequireSessionMfa248102654/002/log/upload/streaming/default: directory not empty
--- FAIL: TestTerminalRequireSessionMfa (11.15s)
```

--- a/lib/web/apiserver_test.go
+++ b/lib/web/apiserver_test.go
@@ -1871,7 +1871,7 @@ func TestTerminalNameResolution(t *testing.T) {
 	}
 }
 
-func TestTerminalRequireSessionMfa(t *testing.T) {
+func TestTerminalRequireSessionMFA(t *testing.T) {
 	ctx := context.Background()
 	env := newWebPack(t, 1)
 	proxy := env.proxies[0]
@@ -7444,10 +7444,18 @@ func newWebPack(t *testing.T, numProxies int, opts ...proxyOption) *webPack {
 			ClusterName: "localhost",
 			Dir:         t.TempDir(),
 			Clock:       clock,
+			AuditLog:    events.NewDiscardAuditLog(),
 		},
 	})
 	require.NoError(t, err)
 	t.Cleanup(func() { require.NoError(t, server.Shutdown(ctx)) })
+
+	// use a sync recording mode because the disk-based uploader
+	// that runs in the background introduces races with test cleanup
+	recConfig := types.DefaultSessionRecordingConfig()
+	recConfig.SetMode(types.RecordAtNodeSync)
+	err = server.AuthServer.AuthServer.SetSessionRecordingConfig(context.Background(), recConfig)
+	require.NoError(t, err)
 
 	// Register the auth server, since test auth server doesn't start its own
 	// heartbeat.

